test/connect.c

```diff
@@ -235,6 +245,102 @@ static int test_connect(struct io_uring *ring)
 	return -1;
 }
 
+static int test_connect_timeout(struct io_uring *ring)
+{
+	int fd = -1, connect_fd = -1, accept_fd = -1;
+	int ret;
+	struct sockaddr_in addr;
+	struct io_uring_sqe *sqe;
+	struct __kernel_timespec ts = {.tv_sec = 0, .tv_nsec = 100000};
+
+	connect_fd = create_socket();
+	if (connect_fd == -1)
+		return -1;
+
+	accept_fd = create_socket();
+	if (accept_fd == -1)
+		goto err;
+
+	if (configure_connect(connect_fd, &addr) == -1)
+		goto err;
+
+	ret = bind(accept_fd, (struct sockaddr*)&addr, sizeof(addr));
+	if (ret == -1) {
+		perror("bind()");
+		goto err;
+	}
+
+	ret = listen(accept_fd, 0);  // no backlog in order to block connect_fd
+	if (ret == -1) {
+		perror("listen()");
+		goto err;
+	}
+
+	// Fill up available place in the accept queue (backlog)
+	fd = create_socket();
+	if (connect(fd, &addr, sizeof(addr)) == -1) {
+			fprintf(stderr, "unable to connect %d\n", errno);
+			goto err;
+	}
+
+	sqe = io_uring_get_sqe(ring);
+	if (!sqe) {
+		fprintf(stderr, "unable to get sqe\n");
+		goto err;
+	}
+
+	io_uring_prep_connect(sqe, connect_fd, (struct sockaddr*)&addr, sizeof(addr));
+	sqe->user_data = 1;
+	sqe->flags |= IOSQE_IO_LINK;
+
+	sqe = io_uring_get_sqe(ring);
+	if (!sqe) {
+		fprintf(stderr, "unable to get sqe\n");
+		goto err;
+	}
+	sqe->user_data = 2;
+
+	io_uring_prep_link_timeout(sqe, &ts, 0);
+	ret = io_uring_submit(ring);
+	if (ret != 2) {
+		fprintf(stderr, "submitted %d\n", ret);
+		return -1;
+	}
+
+	for (int i = 0; i < 2; i++) {
+		int expected;
+		struct io_uring_cqe *cqe;
+
+		ret = io_uring_wait_cqe(ring, &cqe);
+		if (ret) {
+			fprintf(stderr, "wait_cqe=%d\n", ret);
+			return 1;
```

Comment:
umm...
but I think it's good enough and we'll fix it on top. Thanks!

Reply:
no, I am sorry. lack of attention from late hours.